src/repr/tests/strconv.rs

```diff
@@ -15,6 +15,12 @@ use repr::strconv;
 
 #[test]
 fn test_parse_date() {
+    run_test_parse_date("690203", NaiveDate::from_ymd(2069, 2, 3));
+    run_test_parse_date("700203", NaiveDate::from_ymd(1970, 2, 3));
+    run_test_parse_date("010203", NaiveDate::from_ymd(2001, 2, 3));
+    run_test_parse_date("0010203", NaiveDate::from_ymd(1, 2, 3));
+    run_test_parse_date("00010203", NaiveDate::from_ymd(1, 2, 3));
+    run_test_parse_date("20010203", NaiveDate::from_ymd(2001, 2, 3));
     run_test_parse_date("2001-02-03", NaiveDate::from_ymd(2001, 2, 3));
```

Comment:
could you include a couple of negative cases (e.g. from the earlier draft) as well?

something like `20010199` and `-12345678910111213141516171819`.